clients/src/test/java/org/apache/kafka/common/network/NioEchoServer.java

```diff
@@ -226,8 +226,16 @@ public void run() {
                 synchronized (newChannels) {
                     for (SocketChannel socketChannel : newChannels) {
                         String id = id(socketChannel);
-                        selector.register(id, socketChannel);
-                        socketChannels.add(socketChannel);
+                        // This try-catch block prevents the NioEchoServer from crashing in a rare condition
```

Comment:
it seems we don't use the id actually, right? if so, maybe we can use a increment number to be the id to fix id conflicts?

Reply:
Hi @chia7712 , thanks for your feedback.

Ideally, the current method of ID generation should not cause duplication. However, as mentioned in the ticket description, if the channel gets connected and disconnected multiple times in a very short period, it may result in the same ID being assigned to two channels.

To avoid this issue, we could consider using an incrementing ID(as your proposal). 
IMHO, we might consider using UUIDs alternatively, which could simplify the process.
WDYT?

Reply:
How about using a static `AtomicLong`? It should be large enough in our testing.

Reply:
Sounds good 😺

Reply:
FYI, I have looped testing testUngracefulRemoteCloseDuringHandshakeRead and testUngracefulRemoteCloseDuringHandshakeWrite about 500 times each. They work like a charm. 😃